icu4c/source/test/intltest/loctest.cpp

```diff
@@ -3154,3 +3155,43 @@ void LocaleTest::TestBug13417VeryLongLanguageTag() {
     status.errIfFailureAndReset("\"%s\"", l.getName());
     assertEquals("equals", tag, result.c_str());
 }
+
+static const char* const tz_in_ext[] = {
+    "etadd", "tzdar", "eheai", "sttms", "arirj", "arrgl", "aruaq", "arluq",
+    "mxpvr", "brbvb", "arbue", "caycb", "brcgr", "cayzs", "crsjo", "caydq",
+    "svsal", "cafne", "caglb", "cagoo", "tcgdt", "ustel", "bolpb", "uslax",
+    "sxphi", "mxmex", "usnyc", "usxul", "usndcnt", "usndnsl", "ttpos",
+    "brpvh", "prsju", "clpuq", "caffs", "cayek", "brrbr", "mxstis", "dosdq",
+    "brsao", "gpsbh", "casjf", "knbas", "lccas", "vistt", "vcsvd", "cayyn",
+    "cathu", "hkhkg", "mykul", "khpnh", "cvrai", "gsgrv", "shshn", "aubhq",
+    "auldh", "imdgs", "smsai", "asppg", "pgpom",
+};
+static const char* const tzname_with_underline[] = {
+    "America/Buenos_Aires", "America/Coral_Harbour", "America/Los_Angeles",
+    "America/Mexico_City", "America/New_York", "America/Rio_Branco",
+    "America/Sao_Paulo", "America/St_Johns", "America/St_Thomas",
+    "Australia/Broken_Hill", "Australia/Lord_Howe", "Pacific/Pago_Pago"
+};
+
+void LocaleTest::TestBug11053UnderlineTimeZone() {
+    std::string locale_str;
+    for (int32_t i = 0; i < static_cast<int32_t>(sizeof(tz_in_ext) / sizeof(char*)); i++) {
+      locale_str = "en-u-tz-";
+      locale_str += tz_in_ext[i];
+      Locale l(locale_str.c_str());
+      assertTrue((locale_str + " !l.isBogus()").c_str(), !l.isBogus());
+    }
+    for (int32_t i = 0; i < static_cast<int32_t>(sizeof(tzname_with_underline) / sizeof(char*)); i++) {
+      locale_str = "en@timezone=";
```

Comment:
Using a base locale without any separator here means that you here never test that your code correctly parses underscore as part of the base locale, only ignoring it when part of the extension. Using `en_US` instead would test that too.

In fact, instead of using the same base locale for all test cases, using a few different ones, like `en`, `en_US`, `en_Latn_US`, etc., would give much better test coverage.